itemManagement/views.py

```diff
@@ -37,15 +40,67 @@ def get_context_data(self, *args, **kwargs):
 class ItemDetailsView(TemplateView):
     def get(self, request, itemId, *args, **kwargs):
         isAdmin = request.user.is_superuser
+        # isAdmin = True
 
         print(f"Item ID requested: {itemId}")
 
         context = Item.objects.get(id=itemId).getItemDetails()
+
         if isAdmin:
             return render(request, 'itemManagement/item_details_admin.html', context=context)
         else:
             return render(request, 'itemManagement/item_details_assistant.html', context=context)
 
+    def post(self, request, itemId, *args, **kwargs):
+
+        item = Item.objects.get(id=itemId)
+
+        item.lastUsed = timezone.now()
+        # TODO: item lastUsed updated if decrement clicked
+
+        isAdmin = True
+
+        # TODO: Admin validation
+
+        # Admin Fields updating fields other than quantities if admin
+        if isAdmin:
+            # --------TEXT FIELDS--------
+            item.title = request.POST.get('itemName', "").strip()
+            item.description = request.POST.get('description', "").strip()
+            item.price = request.POST.get('price', "").strip()
+            item.unit = request.POST.get('unit', "").strip()
+            item.save(update_fields=['title', 'description', 'price', 'unit'])
+            # ---------------------------
+
+            # --------TAGS-------------
+            newTags = request.POST.get('newTags', "").split(',')
+            # clear all tags
+            item.tag_set.all().delete()
+            # add back all new edited tags
+            for newTag in newTags:
+                if newTag != "" or newTag is None:
+                    newTag = Tag.objects.create(name=newTag.strip(), item=item)
+                    newTag.save()
+            # -------------------------
+
+        # ----Quantities in Locations----
+        # Locations: list of all locations that house the current item
+        locations = item.locations.all()
+
+        # Locations
+        for location in locations:
+            itemStorage = location.itemstorage_set.get(item=item)
+            itemStorage.quantity = request.POST.get('quantity-location-' + str(location.id), "").strip()
+            itemStorage.save()
```

Comment:
Does this work if there is no Item at that location originally? (i.e if itemStorage is null)

Reply:
do you mean if quantity is zero? from, we weren't gonna get rid of entries if the quantity went to zero because then people would have to manually add them if they checked out the entire thing. And "Add item to new location" is in my todo above.

Reply:
When a location or an item is first created there is no item storage relation for them so you will get null values.

Reply:
So in that case, if `locations` is null, then this code just wouldn't execute. and locations rows wouldn't be displayed on the details page if it's null

Reply:
Maybe I'm just missing something. Is there logic elsewhere for if I want to add a location?
User Story
1. Create a new location
2. Choose an item and store 10 units in this new location
What logic handles adding this new location to the item

Reply:
What you're missing is the PR message:

> Still remains:
> - **Add item to new location**
> - Add images
> - Input validation
> - Aesthetic changes

it's not there yet. That's next step along with error handling/testing and eventually images

Reply:
Gotch, my bad

Reply:
All good